### PR TITLE
154208444 Remove operation area facet count from dropdown

### PR DESCRIPTION
**Pull request by rckopteri**
# Added
- Sort transport services operation area drowdown results in a alphabetical order

# Fixed
 - Remove service amount from operation-area-facet dropdown list item value.
 - Delete operation area facets that are created by the users. Leave only finnis_muncipalities areas to operation-area-facet search result

### CODE REVIEW

File: ote/src/clj/ote/services/service_search.sql [hunk 1]
```diff
@@ -1,8 +1,9 @@
 -- name: operation-area-facet
 SELECT "operation-area" as text, COUNT(*) as count
-  FROM "operation-area-facet"
+  FROM "operation-area-facet" o
+ WHERE o."operation-area" IN (SELECT f.namefin FROM "finnish_municipalities" f)
```
Comment (tatut) [inline ote/src/clj/ote/services/service_search.sql]: should we instead do this when the data is generated in `transport_service_operation_area_array` trigger and recreate the full data in a migration

File: ote/src/cljs/ote/views/service_search.cljs [hunk 2]
```diff
@@ -158,7 +158,7 @@
 
         {:name ::t-service/operation-area
          :type :multiselect-selection
-         :show-option #(str (:text %) " (" (:count %) ")")
+         :show-option #(str (:text %))
```
Comment (tatut) [inline ote/src/cljs/ote/views/service_search.cljs]: just use `:text` as show-option function

File: database/src/main/resources/db/migration/V1_45__update_to_service_search_view.sql [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+       ("transport-service-id", "operation-area")
+SELECT t.id, oad.text
+  FROM operation_area oa
+  JOIN "transport-service" t ON oa."transport-service-id" = t.id
+  JOIN LATERAL unnest(oa.description) AS oad ON TRUE
+ WHERE t."published?" = true
+   AND oad.text IN (SELECT f.namefin FROM "finnish_municipalities" f);
+
+-- Create trigger function to update facet when service changes
+CREATE FUNCTION transport_service_operation_area_array () RETURNS TRIGGER AS $$
```
Comment (tatut) [inline database/src/main/resources/db/migration/V1_45__update_to_service_search_view.sql]: CREATE OR REPLACE FUNCTION

File: database/src/main/resources/db/migration/V1_45__update_to_service_search_view.sql [hunk 4]
```diff
@@ -0,0 +1,55 @@
+-- update existing triggers and operation-area-facet table data
+
+-- delete all from operation-area-facet table
+DELETE FROM "operation-area-facet";
+
+-- DROP trigger
+DROP TRIGGER IF EXISTS tg_transport_service_operation_area_array ON "transport-service";
```
Comment (tatut) [inline database/src/main/resources/db/migration/V1_45__update_to_service_search_view.sql]: remove

File: database/src/main/resources/db/migration/V1_45__update_to_service_search_view.sql [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+-- DROP function
+DROP FUNCTION IF EXISTS transport_service_operation_area_array();
```
Comment (tatut) [inline database/src/main/resources/db/migration/V1_45__update_to_service_search_view.sql]: remove

File: database/src/main/resources/db/migration/V1_45__update_to_service_search_view.sql [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+     WHERE oa."transport-service-id" = NEW.id
+       AND oad.text IN (SELECT f.namefin FROM "finnish_municipalities" f);
+  END IF;
+
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Create trigger to update operation area at the end of transaction
+CREATE CONSTRAINT TRIGGER tg_transport_service_operation_area_array
```
Comment (tatut) [inline database/src/main/resources/db/migration/V1_45__update_to_service_search_view.sql]: remove

**Comment by tatut** [comment]
Format description in new PR changelog style

**Comment by K1ll3rF0x** [comment]
Is this good now?

**Comment by rckopteri** [comment]
If opinion count anything, I though that this was ok a week ago.

**Comment by K1ll3rF0x** [comment]
Atleast, there seems to be a name collision in migrations. Please, merge recent master and fix these issues.

**Comment by rckopteri** [comment]
Migration file renamed

**Pull request merged**